### ISSUE TITLE
Refactor documentation generator according json schemas

### ISSUE THREAD

**Issue by krutilin** [beerjson/beerjson#13]
As software developers, we could read existing source code and tests. But most of the people need documentation in its normal form without jumping to code. 

We have barebone of the generator in the docs folder. It should be refactored and it should generate smooth readable documentation.

**Comment by mrichar1** [comment]
Could potentially help with this - though I'm not that much of a JS developer.

Is there any mileage in using a tool like https://github.com/lbovet/docson to do the heavy lifting here?

**Comment by superroma** [comment]
Sorry for later reply
> Is there any mileage in using a tool like https://github.com/lbovet/docson to do the heavy lifting here?
I was hoping to generate a markdown, so I used a version of https://github.com/saibotsivad/json-schema-to-markdown, but it has its issues. 

I have tried docson, and it seems to have problems with our schemas. This one works: https://docson-spgkvoabzj.now.sh/#json/BeerXML.json, but other schemas doesnt: https://docson-spgkvoabzj.now.sh/#json/grain.json
Their hosted environment doesnt work either: http://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/beerjson/beerjson/master/json/BeerXML.json

**Comment by mrichar1** [comment]
I think that the issue in question is this one: https://github.com/lbovet/docson/issues/49 certainly things seemed to mostly work using that fork.

I notice that the docson project has recently become 'undead'  so could be worth following again - though I'm not tied to any particular tool - this just seemed like a potential way to minimise work overall.

**Comment by yanick** [comment]
Hi! I'm the guy who tries to undead docson. Can you check your schemas with the branch `master of https://github.com/yanick/docson ? I've just checked, and with the patch for lbovet/docson#49, `grain.json` seems to be fine. (well, since the main object is an empty object, there is not much to see, but `#/beerjson/json/grain.json$definitions/FermentableBase` looks fine to me)

**Comment by superroma** [comment]
> Can you check your schemas with the branch master of https://github.com/yanick/docson ?

Hi, Your GH pages are served from master, right? I'm trying this URL: http://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/beerjson/beerjson/master/json/grain.json - and it shows me just a box with schema URL and "{}" that expands to schema source.
I tried to clone repo and run server locally (a master branch) - result is the same, here it is deployed: https://docson-uiyzdvmolw.now.sh/#https://raw.githubusercontent.com/beerjson/beerjson/master/json/grain.json

**Comment by superroma** [comment]
Anyway, I started [experimental branch here](https://github.com/beerjson/beerjson/tree/gen-docs-rewrite) to write my own schema-to-md converter. It is [pretty short](https://github.com/beerjson/beerjson/blob/gen-docs-rewrite/js/json-schema-to-markdown.js) so far, thanks to Ramda and ES6 template literals.

**Comment by yanick** [comment]
[GH page] I actually have to check. I've only recently joined the project so I'm not sure. :-)

>  it shows me just a box 

That's what I was alluding to when I said `the main object is an empty object, there is not much to see`. The main object has no attributes and, well, that's exactly what is shown. That's why I looked at one of the definitions -- #/beerjson/json/grain.json$definitions/FermentableBase -- instead.
